python/lsst/ip/diffim/getTemplate.py

```diff
@@ -107,19 +113,32 @@ def run(self, exposure, sensorRef, templateIdList=None):
                 bbox=patchSubBBox,
                 tract=tractInfo.getId(),
                 patch="%s,%s" % (patchInfo.getIndex()[0], patchInfo.getIndex()[1]),
+                numSubfilters=self.config.numSubfilters,
             )
             if patchSubBBox.isEmpty():
                 self.log.info("skip tract=%(tract)s, patch=%(patch)s; no overlapping pixels" % patchArgDict)
                 continue
-            if not sensorRef.datasetExists(**patchArgDict):
-                self.log.warn("%(datasetType)s, tract=%(tract)s, patch=%(patch)s does not exist"
-                              % patchArgDict)
-                continue
 
             nPatchesFound += 1
-            self.log.info("Reading patch %s" % patchArgDict)
-            coaddPatch = sensorRef.get(**patchArgDict)
-            coaddExposure.getMaskedImage().assign(coaddPatch.getMaskedImage(), coaddPatch.getBBox())
+            if self.config.coaddName == 'dcr':
+                if not sensorRef.datasetExists(subfilter=0, **patchArgDict):
+                    self.log.warn("%(datasetType)s, tract=%(tract)s, patch=%(patch)s,"
+                                  " numSubfilters=%(numSubfilters)s, subfilter=0 does not exist"
+                                  % patchArgDict)
+                    continue
+                self.log.info("Constructing DCR-matched template for patch %s" % patchArgDict)
+                dcrModel = DcrModel.fromDataRef(sensorRef, **patchArgDict)
+                coaddPatch = dcrModel.buildMatchedExposure(bbox=patchSubBBox,
+                                                           wcs=coaddWcs,
+                                                           visitInfo=exposure.getInfo().getVisitInfo())
+            else:
+                if not sensorRef.datasetExists(**patchArgDict):
+                    self.log.warn("%(datasetType)s, tract=%(tract)s, patch=%(patch)s does not exist"
+                                  % patchArgDict)
+                    continue
+                self.log.info("Reading patch %s" % patchArgDict)
+                coaddPatch = sensorRef.get(**patchArgDict)
+            coaddExposure.maskedImage.assign(coaddPatch.maskedImage, coaddPatch.getBBox())
```

Comment:
Is this equivalent to `coaddExposure.maskedImage[coaddPatch.getBBox()] = coaddPatch.maskedImage`? If so, that'd be clearer

Reply:
I believe it is equivalent in this instance, but I'd like to leave this as it is for consistency with `assembleCoadd`. If you'd like that changed, I (or someone else) could do that on a separate ticket for both `ip_diffim` and `pipe_tasks`, and possibly other packages as well.